python/cudf/cudf/tests/test_groupby.py

```diff
@@ -1200,3 +1200,41 @@ def test_groupby_nunique_series():
         gdf.groupby("a")["b"].nunique(),
         check_dtype=False,
     )
+
+
+def test_groupby_list_simple():
+    pdf = pd.DataFrame({"a": [1, 1, 1, 2, 2, 2], "b": [1, 2, None, 4, 5, 6]})
+    gdf = cudf.from_pandas(pdf)
+
+    assert_eq(
+        pdf.groupby("a").agg({"b": list}),
+        gdf.groupby("a").agg({"b": list}),
+        check_dtype=False,
+    )
```

Comment:
Should probably test the `collect` API as well just to make sure it works as expected

Reply:
Should we expose that?

Reply:
> Should we expose that?

I'm in favor of exposing it. Using `list` is a bit ambiguous since we're not actually using the Python `list` function anymore, where `collect` is very explicit intention-wise.

Reply:
:+1: done